src/types.go

```diff
@@ -0,0 +1,191 @@
+	IssuesUrl   string `json:"issues_url"`
+	PullsUrl    string `json:"pulls_url"`
+	StatusesUrl string `json:"statuses_url"`
+}
+
+type CheckSuites struct {
+	Count       int          `json:"total_count"`
+	CheckSuites []CheckSuite `json:"check_suites"`
+}
+
+type CheckSuite struct {
+	Id           int                  `json:"id"`
+	HeadBranch   string               `json:"head_branch"`
+	HeadSha      string               `json:"head_sha"`
+	Status       CheckSuiteStatus     `json:"status"`
+	Conclusion   CheckSuiteConclusion `json:"conclusion"`
+	Url          string               `json:"url"`
+	CheckRunsUrl string               `json:"check_runs_url"`
+	App          App                  `json:"app"`
+}
+
+type App struct {
+	Name string `json:"name"`
+}
+
+type Issue struct {
+	Url    string `json:"url"`
+	Number int    `json:"number"`
+	Title  string `json:"title"`
+	State  string `json:"state"`
+}
+
+type IssueComment struct {
+	Url     string `json:"url"`
+	HtmlUrl string `json:"html_url"`
+	Id      int    `json:"id"`
+	Body    string `json:"body"`
+}
+
+type CheckSuiteWebhook struct {
+	Action     ActionType `json:"action"`
+	CheckSuite CheckSuite `json:"check_suite"`
+	Repo       Repo       `json:"repository"`
+}
+
+func IsCheckSuiteSucceeded(conclusion CheckSuiteConclusion) bool {
+	return conclusion == CheckSuiteConclusionSuccess
+}
+
+func IsCheckSuiteFailed(conclusion CheckSuiteConclusion) bool {
+	return conclusion == CheckSuiteConclusionFailure || conclusion == CheckSuiteConclusionTimedOut
+}
+
+func (cs *CheckSuite) IsSucceeded() bool {
+	return IsCheckSuiteSucceeded(cs.Conclusion)
+}
+
+func (cs *CheckSuite) IsFailed() bool {
+	return IsCheckSuiteFailed(cs.Conclusion)
+}
+
+func (csw *CheckSuiteWebhook) IsSucceeded() bool {
+	return csw.CheckSuite.IsSucceeded()
+}
+
+func (csw *CheckSuiteWebhook) IsFailed() bool {
+	return csw.CheckSuite.IsFailed()
+}
+
+func (csw *CheckSuiteWebhook) GetStatusesUrl() string {
+	return strings.ReplaceAll(csw.Repo.StatusesUrl, "{sha}", csw.CheckSuite.HeadSha)
+}
+
+type IssueCommentWebhook struct {
+	Action  ActionType   `json:"action"`
+	Issue   Issue        `json:"issue"`
+	Comment IssueComment `json:"comment"`
+	Repo    Repo         `json:"repository"`
+}
+
+func (pr *PullRequest) GetCheckSuiteUrl() string {
+	return strings.ReplaceAll(pr.Head.Repo.CommitsUrl, "{/sha}", fmt.Sprintf("/%s", pr.Head.Sha)) + "/check-suites"
+}
+
+func (ic *IssueCommentWebhook) GetPullsUrl() string {
+	return strings.ReplaceAll(ic.Repo.PullsUrl, "{/number}", fmt.Sprintf("/%d", ic.Issue.Number))
+}
+
+func NewPullRequest(payload []byte) *PullRequest {
+	var pr PullRequest
+	if err := json.Unmarshal(payload, &pr); err != nil {
+		return nil
+	}
+	if pr.Url == "" && pr.Number == 0 {
+		return nil
+	}
+	return &pr
+}
+
+func NewCheckSuiteWebhook(payload []byte) *CheckSuiteWebhook {
+	var cs CheckSuiteWebhook
+	if err := json.Unmarshal(payload, &cs); err != nil {
+		return nil
+	}
+	if cs.CheckSuite.Id == 0 {
```

Comment:
Would cs.CheckSuite ever be nil?

Reply:
No, see https://github.com/Azure/azure-sdk-actions/pull/1#discussion_r872804863